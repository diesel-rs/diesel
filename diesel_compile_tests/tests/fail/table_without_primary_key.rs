#[macro_use]
extern crate diesel;

table! {
    user {
        //~^ ERROR: neither an explicit primary key found nor does an `id` column exist.
        user_id -> Integer,
        name -> Text,
    }
}

fn main() {}
