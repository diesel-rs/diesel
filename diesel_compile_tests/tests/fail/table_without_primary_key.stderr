error: neither an explicit primary key found nor does an `id` column exist.
       consider explicitly defining a primary key. 
       for example for specifying `user_id` as primary key:
       
       table! {
       
                            user(user_id){
           user (user_id) {
               user_id -> Integer,
               name -> Text,
           }
       }
 --> tests/fail/table_without_primary_key.rs:5:5
  |
LL |     user {
  |     ^^^^
